internal/controller/storageclient_controller.go

```diff
@@ -193,6 +205,46 @@ func (r *StorageClientReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	r.cache = mgr.GetCache()
 	r.crdsBeingWatched.Store(VolumeGroupSnapshotClassCrdName, false)
 
+	if QuotaOpenshiftIoV1ClusterResourceQuota, err = apiutil.GVKForObject(&quotav1.ClusterResourceQuota{}, r.Scheme); err != nil {
+		return err
+	}
+
+	if V1Secret, err = apiutil.GVKForObject(&corev1.Secret{}, r.Scheme); err != nil {
+		return err
+	}
+
+	if NoobaaIoV1Alpha1Noobaa, err = apiutil.GVKForObject(&nbv1.NooBaa{}, r.Scheme); err != nil {
+		return err
+	}
+
+	if StorageV1StorageClass, err = apiutil.GVKForObject(&storagev1.StorageClass{}, r.Scheme); err != nil {
+		return err
+	}
+
+	if SnapshotV1VolumeSnapshotClass, err = apiutil.GVKForObject(&snapapi.VolumeSnapshotClass{}, r.Scheme); err != nil {
+		return err
+	}
+
+	if GroupSnapshotV1Beta1VolumeGroupSnapshotClass, err = apiutil.GVKForObject(&groupsnapapi.VolumeGroupSnapshotClass{}, r.Scheme); err != nil {
+		return err
+	}
+
+	if ReplicationV1Alpha1VolumeReplicationClass, err = apiutil.GVKForObject(&replicationv1a1.VolumeReplicationClass{}, r.Scheme); err != nil {
+		return err
+	}
+
+	if CsiCephIoV1alpha1CephConnection, err = apiutil.GVKForObject(&csiopv1a1.CephConnection{}, r.Scheme); err != nil {
+		return err
+	}
+
+	if CsiCephIoV1alpha1ClientProfile, err = apiutil.GVKForObject(&csiopv1a1.ClientProfile{}, r.Scheme); err != nil {
+		return err
+	}
+
+	if CsiCephIoV1alpha1ClientProfileMapping, err = apiutil.GVKForObject(&csiopv1a1.ClientProfileMapping{}, r.Scheme); err != nil {
+		return err
+	}
```

Comment:
This create more complexity then solves any. 
Looking into the implementation of GVKForObject it seems that we are to run a lot of code that have multiple failure point just to acquire at runtime knowledge we have at build time. 

My suggestion is to revert this change and go back to having each case based on building a GVK on the flay using the GroupVersion and adding the kind based on the type